libcentrifugo/metrics.go

```diff
@@ -109,6 +112,24 @@ type metricsRegistry struct {
 	mu sync.Mutex
 }
 
+func newMetricsHistogramRegistry() *hdrhistogram.HDRHistogramRegistry {
+	quantiles := []float64{50, 90, 99, 99.99}
+	var minValue int64 = 1        // as we record latencies in microseconds, min resolution 1mks
+	var maxValue int64 = 10000000 // as we record latencies in microseconds, max resolution 10s
+	numBuckets := 15              // histograms will be rotated every time we call UpdateSnapshot from outside.
+	sigfigs := 3
+	registry := hdrhistogram.NewHDRHistogramRegistry()
+	registry.Register(hdrhistogram.NewHDRHistogram("http_api", numBuckets, minValue, maxValue, sigfigs, quantiles))
+	registry.Register(hdrhistogram.NewHDRHistogram("client_api", numBuckets, minValue, maxValue, sigfigs, quantiles))
```

Comment:
I didn't think too hard.

I was most interested in whether we can expose the difference between say publishing stuff and fetching history/messages etc.

It's not crucial though.

One option I was considering (ignoring the aggregated part still) would be that we start a timer before JSON unmarshall, but then don't record the value until the end of the request (we do that here already with defer...) and then by the time we record it, we know the command(s) issues and can record them like `http_publish`. If JSON fails we can record it as `http_error` or something?

The alternative would be to measure individual commands like you said as well as overall requests like now - that way you get some insight into which operations are fast/slow but also what general overhead of Marshalling is?

I dunno now I'm thinking about it it's harder to see the value. I was hoping that this would be detailed enough that anyone could do performance testing without instrumenting - e.g. you'd be able to do next round of optimisations of publish path just using these metrics rather than external measurements or additional logging in the code.

But maybe that much detail is not generally useful.

This PR is a great enhancement at any rate Maybe lets push it like this and we can see if additional detail is useful to anyone later?